src/react_native/hooks.cljs

```diff
@@ -10,3 +12,21 @@
 (defn use-back-handler
   [handler]
   (.useBackHandler hooks handler))
+
+(defn use-interval
+  [cb cleanup-cb delay]
+  (let [saved-callback (rn/use-ref)]
+    (rn/use-effect
+     (fn []
+       (oops/oset! saved-callback "current" cb))
+     [cb])
```

Comment:
Just a question, I'm not very used to write code using `use-effect`:

why do we are storing the callback using a ref? this `use-effect` is looking at callback changes, so I think it'd be the same if we just don't store it, but I'm not sure.

Another related question is, what will the differences be (in terms of behaviour) if we used this code instead of two effects:

```clojure
(rn/use-effect
 (fn []
   (when-not (= delay nil)
     (let [id (js/setInterval cb delay)]
       (fn []
         (cleanup-cb)
         (js/clearInterval id)))))
 [cb delay])
 ```

Reply:
> Just a question, I'm not very used to write code using `use-effect`:
> 
> why do we are storing the callback using a ref? this `use-effect` is looking at callback changes, so I think it'd be the same if we just don't store it, but I'm not sure.
> 
> Another related question is, what will the differences be (in terms of behaviour) if we used this code instead of two effects:
> 
> ```clojure
> (rn/use-effect
>  (fn []
>    (when-not (= delay nil)
>      (let [id (js/setInterval cb delay)]
>        (fn []
>          (cleanup-cb)
>          (js/clearInterval id)))))
>  [cb delay])
> ```

hmm, so I will be clear in that I took this code from an old app I had written a while back. It's been some time since I thought through it.

I played around with both approaches for a while and from what I can see using the two use-effects allows the tick callback seamlessly be changed without affecting the timer. I don't particularly need it for this use case but I guess it's better to keep this way since it makes not so much difference. The use-ref is there as changing a ref does not rerender the screen (in case you were not aware of that already).

Reply:
Ok! Thank you so much!! :)